packages/design-system/src/atoms/buttonV2/index.tsx

```diff
@@ -116,75 +117,85 @@ const BaseButton = ({
+            </View>
+          )}
+          {!iconVariant && icon && variant !== 'text' && (
+            <View style={[styles.iconContainer]}>{icon}</View>
+          )}
+          {variant !== 'icon' && (
+            <TypographyV2
+              variant={fontBasedOnVariantAndSize}
+              selectable={false}
+              color={stylesBasedOnState.textColor}
+              style={[styles.title]}
+            >
+              {title}
+            </TypographyV2>
+          )}
+        </>
+      )}
+    </Pressable>
+  )
   return (
-    <View style={[container.alignCenter, style]}>
-      <Pressable
-        testID={testID}
-        isDisabled={isLoading || isDisabled}
-        onHoverIn={() => setIsHover(true)}
-        onHoverOut={() => setIsHover(false)}
-        onPressIn={() => setIsActive(true)}
-        onPressOut={() => setIsActive(false)}
-        onPress={onPress}
-        isHovered={isHover}
-        // Styles
-        position="relative"
-        alignItems="center"
-        justifyContent="center"
-        paddingX={`${paddingX}px`}
-        paddingY={`${paddingY}px`}
-        borderRadius={borderRadius}
-        borderColor={stylesBasedOnState.borderColor}
-        borderWidth={rest.borderWidth}
-        backgroundColor={
-          iconBackgroundColor || stylesBasedOnState.backgroundColor
-        }
-        flexDirection="row"
-        _hover={{
-          borderColor: hover.borderColor,
-          borderWidth: hover.borderWidth,
-        }}
-        _focus={{
-          backgroundColor: iconBackgroundColor || active.backgroundColor,
-        }}
-        _pressed={{
-          borderColor: active.borderColor,
-          borderWidth: active.borderWidth,
-          backgroundColor: iconBackgroundColor || active.backgroundColor,
-        }}
-      >
-        {isLoading ? (
-          <Spinner size={lineHeight} color={stylesBasedOnState.textColor} />
-        ) : (
-          <>
-            {iconVariant && variant !== 'text' && (
-              <View style={[styles.iconContainer]}>
-                <Icon
-                  variant={iconVariant}
-                  width={iconBasedOnSize[size]}
-                  height={iconBasedOnSize[size]}
-                  color={iconColor || stylesBasedOnState.textColor}
-                />
-              </View>
-            )}
-            {!iconVariant && icon && variant !== 'text' && (
-              <View style={[styles.iconContainer]}>{icon}</View>
-            )}
-            {variant !== 'icon' && (
-              <TypographyV2
-                variant={fontBasedOnVariantAndSize}
-                selectable={false}
-                color={stylesBasedOnState.textColor}
-                style={[styles.title]}
-              >
-                {title}
-              </TypographyV2>
-            )}
-          </>
-        )}
-      </Pressable>
+    
+    <View style={[container.alignCenter, style]}> 
+    {variant === 'icon' ?
+    <TooltipV2 title={title}>{button}</TooltipV2>
```

Comment:
I think `title` is not the prop we have to pass here. `title` relates to the Button's text. We have to allow add any text to the Tooltip. You've already added `tooltip` prop in the `types.ts`. I think you need to use it.

```suggestion
    <TooltipV2 title={tooltip}>{button}</TooltipV2>
```